superset/assets/src/components/FilterableTable/FilterableTable.jsx

```diff
@@ -152,21 +153,32 @@ export default class FilterableTable extends PureComponent {
     const widthsByColumnKey = {};
     this.props.orderedColumnKeys.forEach((key) => {
       const colWidths = this.list
-        .map(d => getTextWidth(d[key]) + PADDING) // get width for each value for a key
-        .push(getTextWidth(key) + PADDING); // add width of column key to end of list
+        // get width for each value for a key
+        .map(d => getTextWidth(
+          this.getCellContent({ cellData: d[key], columnKey: key })) + PADDING,
+        )
+        // add width of column key to end of list
+        .push(getTextWidth(key) + PADDING);
       // set max width as value for key
       widthsByColumnKey[key] = Math.max(...colWidths);
     });
     return widthsByColumnKey;
   }
 
-  fitTableToWidthIfNeeded() {
-    const containerWidth = this.container.clientWidth;
-    if (this.totalTableWidth < containerWidth) {
-      // fit table width if content doesn't fill the width of the container
-      this.totalTableWidth = containerWidth;
+  getCellContent({ cellData, columnKey }) {
+    const complexColumn = this.props.expandedColumns.some(
+      name => name.startsWith(columnKey + '.'),
+    );
+    const content = String(cellData);
+    let type;
+    if (content.substring(0, 1) === '[') {
```

Comment:
Not sure if this is over optimization at this point, but since this function could be called tens of thousands of times, it might be better to set `content.substring(0,1)` to a const instead of calling it twice.